parity/run.rs

```diff
@@ -80,27 +81,37 @@ pub struct RunCmd {
 }
 
 pub fn execute(cmd: RunCmd) -> Result<(), String> {
-	// increase max number of open files
-	raise_fd_limit();
-
-	// set up logger
-	let logger = try!(setup_log(&cmd.logger_config));
-
 	// set up panic handler
 	let panic_handler = PanicHandler::new_in_arc();
 
+	// increase max number of open files
+	raise_fd_limit();
+
 	// create dirs used by parity
 	try!(cmd.dirs.create_dirs());
 
 	// load spec
 	let spec = try!(cmd.spec.spec());
+
+	// user defaults path
+	let user_defaults_path = cmd.dirs.user_defaults_path(&spec.fork_name);
+
+	// load user defaults
+	let mut user_defaults = try!(UserDefaults::load(&user_defaults_path));
+
+	// check if tracing is on
+	let tracing = try!(tracing_switch_to_bool(cmd.tracing, &user_defaults));
+
+	// set up logger
+	let logger = try!(setup_log(&cmd.logger_config));
```

Comment:
Why logger is moved here? I think it would be better to initialize logger as a first thing.